### ISSUE TITLE
Sending a test event fails with in `:proplists.get_value/3`

### ISSUE THREAD

**Issue by rlopzc** [getsentry/sentry-elixir#788]
### Environment

What version are you running? 10.7.1

### Steps to Reproduce

1. Configure prod
2. run `MIX_ENV=prod mix sentry.send_test_event`

### Expected Result

Test event is sent.

### Actual Result

An error is raised in the console.

```
Client configuration:
server: REDACTED/envelope/
public_key: REDACTED
secret_key:
current environment_name: "prod"
hackney_opts: [pool: :sentry_pool]

Sending test event...
** (FunctionClauseError) no function clause matching in :proplists.get_value/3

    The following arguments were given to :proplists.get_value/3:

        # 1
        "X-Sentry-Error"

        # 2
        %ArgumentError{message: "unknown registry: Req.Finch"}

        # 3
        nil

    (stdlib 6.0.1) proplists.erl:214: :proplists.get_value/3
    (sentry 10.7.1) lib/sentry/client.ex:386: Sentry.Client.maybe_log_send_result/2
    (sentry 10.7.1) lib/sentry/client.ex:251: Sentry.Client.encode_and_send/4
    (sentry 10.7.1) lib/sentry/client.ex:137: Sentry.Client.send_event/2
    (sentry 10.7.1) lib/mix/tasks/sentry.send_test_event.ex:88: Mix.Tasks.Sentry.SendTestEvent.send_event/1
    (mix 1.17.2) lib/mix/task.ex:495: anonymous fn/3 in Mix.Task.run_task/5
    (mix 1.17.2) lib/mix/cli.ex:96: Mix.CLI.run_task/2
    /Users/rlopez/.asdf/installs/elixir/1.17.2-otp-27/bin/mix:2: (file)
```

My config:
```
config :sentry,
  client: Gru.Sentry.HttpClient,
  release: System.get_env("APP_VERSION", "0.0.1"),
  dsn: "REDACTED",
  environment_name: config_env(),
  enable_source_code_context: true,
  root_source_code_paths: [File.cwd!()],
  report_deps: false,
  integrations: [
    oban: [
      # Capture errors:
      capture_errors: true,
      # Monitor cron jobs:
      cron: [enabled: true]
    ]
  ]
```

My HTTP client:
```elixir
defmodule App.Sentry.HttpClient do
  @moduledoc """
  Sentry HTTPClient using Req
  """
  @behaviour Sentry.HTTPClient

  @impl Sentry.HTTPClient
  def post(url, request_headers, request_body) do
    request_headers = Map.new(request_headers)

    case Req.post(url, headers: request_headers, body: request_body) do
      {:ok, %Req.Response{status: status, headers: headers, body: body}} ->
        {:ok, status, headers, body}

      {:error, error} ->
        {:error, error}
    end
  end
end
```

**Comment by whatyouhide** [comment]
@savhappy can you take a look at this?

**Comment by whatyouhide** [comment]
@savhappy yeah I think we should prevent this kind of deep-within-the-lib errors when possible. In this case, I think we should not assume that we have well-formed headers in `Sentry.Client.maybe_log_send_result/2` and instead check for that. Let's make sure to add a test for this regression too 🙃

**Comment by savhappy** [comment]
Okay, so what's actually happening here is that an error exception is being returned. Which is great because it tells us exactly what's wrong (Req is returning a struct so the body isn't encoded, thus the Jason.decode fails). However, I only noticed this after I commented out the `:proplists.get_value()` function and let the entire error fall through. 

I think we should allow this error exception through on the `Sentry.Client.maybe_log_send_result/2` like it is in `Sentry.Client.send_event/2`.  This is in PR [#789](https://github.com/getsentry/sentry-elixir/pull/789)

```
%Sentry.ClientError{reason: :server_error, http_response: {:error, :badarg, [{:erlang, :iolist_to_binary, [%{"id" => "49fd6554bb7f491e8956324a8071802e"}], [error_info: %{module: :erl_erts_errors}]}, {Jason, :decode, 2, [file: ~c"lib/jason.ex", line: 69]}, {Sentry.Transport, :request, 4, [file: ~c"lib/sentry/transport.ex", line: 62]}, {Sentry.Transport, :post_envelope_with_retries, 5, [file: ~c"lib/sentry/transport.ex", line: 34]}, {Sentry.Client, :encode_and_send, 4, [file: ~c"lib/sentry/client.ex", line: 249]}, {Sentry.Client, :send_event, 2, [file: ~c"lib/sentry/client.ex", line: 137]}, {Mix.Tasks.Sentry.SendTestEvent, :send_event, 1, [file: ~c"lib/mix/tasks/sentry.send_test_event.ex", line: 88]}, {Mix.Task, :"-run_task/5-fun-3-", 3, [file: ~c"lib/mix/task.ex", line: 495]}]}}

```

@whatyouhide I'm making another PR to handle header validation because this issue might arise again for `Req` users.

**Comment by whatyouhide** [comment]
This has been fixed in #794.
